### PR TITLE
Allow creating snapshot from VM snapshot

### PR DESCRIPTION
**Pull request by ravening**
In old UI we support creating snapshot from
VM snapshot by selecting one of the volume
belonging to VM. Allow the same in new UI

Navigate to Storage -> VM Snapshots -> Select any one -> Click on camera icon which says create snapshot

Screenshot 2020-08-17 at 18 05 25

### CODE REVIEW

**Comment by rhtyd** [comment]
@davidjumani can you review this

**Comment by rhtyd** [comment]
@blueorangutan package

File: src/config/section/storage.js
Comment (davidjumani) [inline src/config/section/storage.js:336]: @ravening Could you add a check whether the instance hypervisor is KVM since the legacy UI only supports it for KVM.
https://github.com/apache/cloudstack/blob/master/ui/scripts/storage.js#L3220

Comment (ravening) [reply src/config/section/storage.js:336]: @davidjumani I dont find the key `hypervisor` in `record`. So it always returns undefined when I try to access `record.hypervisor`. Any idea on this?

Comment (davidjumani) [reply src/config/section/storage.js:336]: @ravening The hypervisor can be found via the `virtualmachineid` in the record

Comment (rhtyd) [reply src/config/section/storage.js:336]: @ravening @davidjumani can we make it simpler then? maybe change the list response to pass hypervisor and use it here. Note: 1.0/GA ships with 4.15 - so if it makes it simpler, send a PR against 4.14 branch and return the key as part of vmsnapshot list response?

Comment (ravening) [reply src/config/section/storage.js:336]: @rhtyd if we have to return hypervisor in list snapshot response then we need to call listvirtualmachine api for each snpashot response which might increase response time and also burden on mgt server


As @davidjumani suggested, we have virtulamachineid parameter but we need to call listvirtualmachine api to get the hypervisor type. Im trying to make that work

Comment (davidjumani) [reply src/config/section/storage.js:336]: @ravening That might not work since api calls are async and will return after the `show` method has completed execution. It would be better to alter the API response in the backend to include the vm's hypervisor which could be done quite easily without calling further apis.

Comment (ravening) [reply src/config/section/storage.js:336]: @davidjumani @rhtyd so then I will just check `record.hypervisor === 'KVM'` here and add `hypervisor` field to listVMSnapshot response in backend.

I still need to call list virtual machine to get the hypervisor type

Comment (davidjumani) [reply src/config/section/storage.js:336]: @ravening Yes, add that condition, once the backend has been changed to return the hypervisor as well, we can merge this!
There won't be a need to call another API as the hypervisor field will be added in the backend itself

Comment (ravening) [reply src/config/section/storage.js]: @davidjumani I have updated the code here. In the meantime I will work on adding hypervisor type in the backend code

Comment (davidjumani) [reply src/config/section/storage.js]: @ravening `record.state === 'Ready' && record.hypervisor === 'KVM'` :smile:

**Comment by rhtyd** [comment]
LGTM, wil merge this as soon as we merge the upstream 4.14 PR https://github.com/apache/cloudstack/pull/4275

**Review by rhtyd — APPROVED** [review]

**Comment by rhtyd** [comment]
@ravening is this still WIP, or ready?

**Comment by ravening** [comment]
> @ravening is this still WIP, or ready?

@rhtyd this is ready.. waiting for back end pr to get merged

**Pull request merged**
